src/compiler/checker.ts

```diff
@@ -5750,6 +5755,27 @@ namespace ts {
                 }
                 return Ternary.False;
             }
+
+            function enumRelatedTo(source: Type, target: Type) {
+                if (source.symbol.name !== target.symbol.name ||
+                    source.symbol.flags & SymbolFlags.ConstEnum ||
+                    target.symbol.flags & SymbolFlags.ConstEnum) {
+                    return Ternary.False;
+                }
+                const sourceDecl = <EnumDeclaration>getDeclarationOfKind(source.symbol, SyntaxKind.EnumDeclaration);

```

Comment:
well, you already have types on hand, so you can get its members that has flag `SymbolFlag.EnumMember`. Open question is if you need to pay attention to the order of member since it can directly affect values of enum members